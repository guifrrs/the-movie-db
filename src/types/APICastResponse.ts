export type APICastResponse = {
  id: number
  cast: {
    id: number,
    name: string,
    profile_path: string,
    character: string,
  }[]
  crew: {
    id: number,
    name: string,
    profile_path: string,
    job: string,
  }[]
}