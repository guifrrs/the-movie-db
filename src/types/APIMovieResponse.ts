export type APIMovieResponse = {
  id: number
  title: string
  poster_path: string
  release_date: string
  genres: {
    id: number
    name: string
  }[]
  overview: string
  runtime: number
  vote_average: number
  vote_count: number
  credits: {
    cast: {
      id: number
      name: string
      profile_path: string
      character: string
    }[]
    crew: {
      id: number
      name: string
      profile_path: string
      job: string
    }[]
  }
  videos: {
    results: {
      id: string
      key: string
      name: string
      site: string
      type: string
    }[]
  }
}