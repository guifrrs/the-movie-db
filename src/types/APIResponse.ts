export interface APIResponse {
  page: number
  results: {
    id: number,
    title: string,
    poster_path: string,
    release_date: string,
  }[]
  total_pages: number
  total_results: number
}