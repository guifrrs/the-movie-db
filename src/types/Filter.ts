import { FILTERS_VALUES } from "@/utils";

export type Filter = (typeof FILTERS_VALUES)[number]