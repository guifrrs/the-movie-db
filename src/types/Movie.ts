export type Movie = {
  id: number
  title: string
  poster: string
  releaseDate: string
  genres: string[]
  description: string
  runtime: number
}