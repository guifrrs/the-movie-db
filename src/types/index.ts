export * from './APIPopularResponse'
export * from './APICastResponse'
export * from './APIVideoResponse'
export * from './Movie'
export * from './Filter'