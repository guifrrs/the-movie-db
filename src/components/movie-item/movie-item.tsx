import Link from 'next/link'
import Image from 'next/image'
import { API, formatDate } from '@/utils'
import { Movie } from '@/types'

interface MovieItemProps {
  movie: Movie
}

export function MovieItem({ movie }: MovieItemProps) {
  const { title, poster, releaseDate } = movie
  const moviePoster = API.POSTER.concat(poster)

  return (
    <>
      <Link href="/">
        <Image src={`${moviePoster}`} alt={`Poster for ${title}`} className="mb-3" width={300} height={300} />
      </Link>

      <h3 className="text-black text-sm font-bold leading-tight">{title}</h3>
      <p className="text-stone-500 text-xs font-bold leading-none">{formatDate(releaseDate)}</p>
    </>
  )
}
