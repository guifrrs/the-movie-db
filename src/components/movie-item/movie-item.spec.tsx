import { Movie } from '@/types'
import { render, screen } from '@testing-library/react'
import { MovieItem } from './movie-item'

const defaultProps: Movie = {
  id: 1,
  title: 'Some title',
  poster: '/poster.jpg',
  releaseDate: '2021-05-22 00:00:00',
}

describe('MovieItem', () => {
  it('render the movie item', () => {
    const { container } = render(<MovieItem movie={defaultProps} />)

    expect(screen.getByText(/some title/i)).toBeInTheDocument()
    expect(screen.getByText(/may 22, 2021/i)).toBeInTheDocument()
    expect(screen.getByAltText(/poster for some title/i)).toBeInTheDocument()

    expect(container).toMatchSnapshot()
  })
})
