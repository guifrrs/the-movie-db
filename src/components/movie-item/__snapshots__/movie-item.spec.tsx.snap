// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieItem render the movie item 1`] = `
<div>
  <a
    href="/"
  >
    <img
      alt="Poster for Some title"
      class="mb-3"
      data-nimg="1"
      decoding="async"
      height="300"
      loading="lazy"
      src="/_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw300%2Fposter.jpg&w=640&q=75"
      srcset="/_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw300%2Fposter.jpg&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw300%2Fposter.jpg&w=640&q=75 2x"
      style="color: transparent;"
      width="300"
    />
  </a>
  <h3
    class="text-black text-sm font-bold leading-tight"
  >
    Some title
  </h3>
  <p
    class="text-stone-500 text-xs font-bold leading-none"
  >
    May 22, 2021
  </p>
</div>
`;
